### PR TITLE
Support 'word-spacing' for geckolib

### PR DESCRIPTION
**Pull request by ghost**
**These changes haven't been tested yet!**

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] There are tests for these changes

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12795)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1213,6 +1213,21 @@ fn static_assert() {
...
+    pub fn set_word_spacing(&mut self, v: longhands::word_spacing::computed_value::T) {
+        use values::computed::LengthOrPercentage::*;
+
+        match v.0 {
+            Some(lop) => match lop {
+                Length(au) => self.gecko.mWordSpacing.set_value(CoordDataValue::Integer(au.0)),
+                Percentage(f) => self.gecko.mWordSpacing.set_value(CoordDataValue::Percent(f)),
+                Calc(l_p) => self.gecko.mWordSpacing.set_value(CoordDataValue::Calc(l_p.into())),
+            },
+            None => self.gecko.mWordSpacing.set_value(CoordDataValue::Auto),
```
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: None should be zero I think. Or None. The spec says 0 but we might be able to get away with using None if gecko layout treats it as the same.

File: components/style/properties/gecko.mako.rs [hunk 2]
```diff
@@ -1213,6 +1213,21 @@ fn static_assert() {
 
     <%call expr="impl_coord_copy('line_height', 'mLineHeight')"></%call>
 
+    pub fn set_word_spacing(&mut self, v: longhands::word_spacing::computed_value::T) {
+        use values::computed::LengthOrPercentage::*;
+
+        match v.0 {
+            Some(lop) => match lop {
+                Length(au) => self.gecko.mWordSpacing.set_value(CoordDataValue::Integer(au.0)),
```
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: I think this should be `CoordDataValue::Coord`

File: components/style/properties/gecko.mako.rs (hunk 1 above)
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: Yes, treat this the same as `Length(0)`

File: components/style/properties/gecko.mako.rs (hunk 2 above)
Comment (Manishearth) [inline components/style/properties/gecko.mako.rs]: Yes, it should be coord.

**Comment by Wafflespeanut** [comment]
Comments make sense. I've made the changes :)

**Comment by Manishearth** [comment]
@bors-servo r+

This works.

screen shot 2016-08-11 at 5 46 37 pm

**Comment by Manishearth** [comment]
@bors-servo retry

 - github doesn't like us

**Comment by Manishearth** [comment]
@bors-servo retry

 - https://github.com/servo/servo/issues/12818

**Comment by Manishearth** [comment]
@bors-servo retry

hmm

**Comment by Manishearth** [comment]
@bors-servo retry

 - #8769

**Pull request merged**